### DISCUSSION TITLE
Manually set locale

### DISCUSSION THREAD

**Discussion by bgdev89** [amannn/next-intl#320] [Q&A] [answered]
Is there a way to manually set the locale when using the useTranslations hook? As far as I can tell the `getLocale` method is responsible for determining the locale, but there is no override. 

Right now, I'm manually setting the request header in middleware, but it feels a bit hacky. Is there a better way to handle this?

**Comment by amannn** [comment, marked as answer]
Thank you for the question! I've just added added improved support for this use case, please see https://github.com/amannn/next-intl/issues/408#issuecomment-1637492614
